base/inference.jl

```diff
@@ -308,6 +306,71 @@ function get_staged(li::MethodInstance)
 end
 
 
+mutable struct OptimizationState
+    linfo::MethodInstance
+    vararg_type_container #::Type
+    backedges::Vector{Any}
+    src::CodeInfo
+    mod::Module
+    nargs::Int
+    next_label::Int # index of the current highest label for this function
+    min_valid::UInt
+    max_valid::UInt
+    params::InferenceParams
+    function OptimizationState(frame::InferenceState)
+        s_edges = frame.stmt_edges[1]
+        if s_edges=== ()
+            s_edges = []
+            frame.stmt_edges[1] = s_edges
```

Comment:
I could make it part of the initialization for frame, but it's just more sensible to handle the lazy initialization here.